### DISCUSSION TITLE
Cannot run both api/ and app/api functions locally at the same time

### DISCUSSION THREAD

**Discussion by btahir** [vercel/community#3232] [Bug Report]
### Page to Investigate

NA

### Steps to Reproduce

Hi - 

I have a project using the new app directory with some api routes in app/api. I needed some python specific functions so have them on the root level in api/ folder. My vercel.json is configured to route these using the routes key. For example:

```
{
  "routes": [
    {
      "src": "/api/hello",
      "dest": "api/hello.py"
    },
...
  ]
}
```

This all works fine on Vercel but locally if I run pnpm dev I can only access the app/api/* functions. If I run vercel dev I can access the python functions in api/*. But I can't get both?

**Comment by ralphsmith80** [comment]
I'm having the same issue. I noticed the [nextjs-flask](https://github.com/vercel/examples/blob/main/python/nextjs-flask/README.md) example is close to what I want and might be what I eventually need, but it seems like an unnecessary step right now. 

I think the key is that this example runs two servers. A `next-dev` and `flask-dev` server. The `flask-dev` server runs on port `5238`, which you can see in the [package.json](https://github.com/vercel/examples/blob/main/python/nextjs-flask/package.json#L6) file and allows the connection described in the README. 
> The Flask server will be running on http://127.0.0.1:5328/ – feel free to change the port in package.json (you'll also need to update it in next.config.js).

I don't yet see a way to do this without having a python server as the single-entry point locally. I'm guessing if you don't have any functions in `app/api/` the next dev server is smart enough to wrap he function in a lightweight python server or maybe there's a single instance that wraps them as routes. I'm not seeing how they do that in the repo. 

The example doesn't have any `/app/api` files, but I added one and it works. 

In that setup I think this would be the main thing to be aware:
> In production, the Flask server is hosted as [Python serverless functions](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/python) on Vercel.

You'll end up with one serverless function and now you need to be aware that your routes are not based on files, but the defined Flask routes. The actual python API URL to get the hello response in the example is http://localhost:5328/api/python as opposed to http://localhost:5328/api/index.

**Reply by ralphsmith80** [reply]
Just a note. You can have individual Python functions, a Flask server function, and node functions, but I had to put the Python functions all at the same level. Even with rewrite rules I couldn't nest the Flask server function in a directory. `/api/index.py` is the Flask server function.

![image](https://github.com/vercel/community/assets/1655592/4b27abfe-4002-4f9a-8817-a54c75b4d71a)

**Reply by btahir** [reply]
Thank you for the deep dive! At least I understand the issue much better now. 🙏 😄

**Reply by nachaos** [reply]
Having the same issue now. Is there a working example for this? :)
